.github/workflows/linux.yml

```diff
@@ -0,0 +1,51 @@
+name: "Linux builds"
+on: push
+jobs:
+
+  build_ubuntu_gcc9_debug:
+    name: "GCC 9 debug (ubuntu-18.04)"
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - run: sudo ./scripts/build.sh --install-deps --compiler-version 9 --release debug
```

Comment:
I agree, but didn't have a more elegant way to bundle the two together.  One thought is to switch the action itself (--install-deps) into instructive output (--show-deps), and it would simply print the apt get install <packages> for the user to execute on their own terms.  

This would let us leverage it inside the YAML with something like:  sudo apt install $(./build.sh --show-deps), and the build script becomes more part of the code (so changes to our libs could change yet the CI YAML would stay the same)

Reply:
I like this idea a lot - it's similar to how pkgconfig works - also as I just discovered, it could be easily reused for injecting build dependencies for rpm packages starting with Fedora 31. How about one more step and moving dependency printing into a separate script?
`sudo apt install $(./scripts/list-build-dependencies.sh)`?

Reply:
Nice and tidy; will do!
The expansion approach works great under brew and apt, but I can't the nested shell expansion to work properly in Windows. 

Here's what I've tried so far (in the YAML):

`run: C:\tools\msys64\usr\bin\bash -lc "pacman -S --noconfirm $($GITHUB_WORKSPACE/scripts/build.sh --list-packages --bin-path /mingw64/bin)"`

That threw syntax errors on the first $, so I decided to switch to the (not-recommended) back-tick sub-shell:

```
run: C:\tools\msys64\usr\bin\bash -lc "pacman -S --noconfirm `$GITHUB_WORKSPACE/scripts/build.sh --list-packages --bin-path /mingw64/bin`"
```

But that also failed.

I gave up so far and have explicitly listed the packages.  Need to figure something out.

Reply:
Windows uses cmd.exe as default shell, it can be changed to PowerShell or bash.

Reply:
Example:

    steps:
      - name: Display the path
        run: echo $PATH
        shell: bash

Link to documentation: https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsrun